src/rgw/rgw_op.h

```diff
@@ -1773,7 +1773,8 @@ class RGWGetObjLayout : public RGWOp {
   rgw_raw_obj head_obj;
 
 public:
-  RGWGetObjLayout() {
+  RGWGetObjLayout() = default;
+  ~RGWGetObjLayout() {
```

Comment:
As I checked quickly, the member variable target is only used inside the execute() function. So no need of a member variable. Just use `std::unique_ptr<RGWRados::Object> target;` locally inside execute() with new operator and without a delete. That should be fine, unless the target variable is used somewhere else later.